dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/rmw/RMWStoreExclusive.java

```diff
@@ -11,26 +11,38 @@
 
 public class RMWStoreExclusive extends Store {
 
-    public RMWStoreExclusive(IExpr address, ExprInterface value, String mo, boolean strong) {
+    private final boolean isStrong;
+    private final boolean requiresMatchingAddresses;
+
+    public RMWStoreExclusive(IExpr address, ExprInterface value, String mo,
+                             boolean isStrong, boolean requiresMatchingAddresses) {
         super(address, value, mo);
         addFilters(Tag.EXCL, Tag.RMW);
-        if (strong) {
+        this.isStrong = isStrong;
+        this.requiresMatchingAddresses = requiresMatchingAddresses;
+        if (isStrong) {
             addFilters(Tag.STRONG);
         }
     }
 
     protected RMWStoreExclusive(RMWStoreExclusive other) {
         super(other);
+        this.isStrong = other.isStrong;
+        this.requiresMatchingAddresses = other.requiresMatchingAddresses;
     }
 
+    public boolean isStrong() { return this.isStrong; }
+    public boolean doesRequireMatchingAddresses() { return this. requiresMatchingAddresses; }
+
     @Override
     public boolean cfImpliesExec() {
-        return is(Tag.STRONG); // Strong RMWs always succeed
+        return isStrong; // Strong RMWs always succeed
     }
 
     @Override
     public String toString() {
-        String tag = is(Tag.STRONG) ? " strong" : "";
+        String tag = isStrong ? " strong" : "";
+        tag += requiresMatchingAddresses ? " addrmatch" : "";
```

Comment:
Do you think this one is really useful?

Reply:
The extra "addrmatch" tag here? It's only useful for debugging in case you accidentally generate the wrong one's for some architecture. I can remove it again if you want.